tests/unit/s2n_connection_test.c

```diff
@@ -61,7 +61,7 @@ int main(int argc, char **argv)
      */
     {
         /* Carefully consider any increases to this number. */
-        const uint16_t max_connection_size = 18700;
+        const uint16_t max_connection_size = 19552;
```

Comment:
Why would the connection struct size be affected by the Round 3 PQ PR? I don't see any changes related to the connection itself.

Reply:
Looks like we're storing an array of `s2n_kem_group_params` for each supported kem group? That seems like a waste of space.

https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/tls/s2n_crypto.h#L42

Reply:
This line increased the maximum number of supported KEMS, which is used to define some array sizes: https://github.com/aws/s2n-tls/pull/2842/files#diff-94a958fa23708fa31b038adc29db44492adfe06b3932bba4bc822bd8262ff8adR85

Reply:
I think the correct fix here is to allocate the array if needed and just store the pointer on the connection.

Reply:
Still getting some failures:
```
s2n_connection size (19584) no longer in (14664, 19552). Please verify that this change was intentional and then update this test. (s2n_connection_test.c line 76)
```

Reply:
Opened #2935 to followup on the memory increase